bladex/propeller.py

```diff
@@ -10,62 +10,83 @@
     """
 
-    def __init__(self, shaft, blade, number_of_blades):
-        self.shaft = shaft
-        self.blade = blade
-        self.number_of_blades = number_of_blades
-
-    def generate_propeller(self, propeller_and_shaft):
-        """
-        Generate and export the .iges and .stl CAD for the propeller with shaft.
-        This method requires PythonOCC (7.4.0) to be installed.
-
-        :raises RuntimeError: if the solid assembling of blades is not 
-            completed successfully
-        :param string propeller_and_shaft: path where to store the .iges and 
-            .stl CAD for the propeller and shaft
-        """
-        shaft_solid = self.shaft.generate_shaft_solid()
-        self.blade.apply_transformations(reflect=True)
-        blade_solid = self.blade.generate_blade_solid(max_deg=2,
+    def __init__(self, shaft, blade, n_blades):
+        self.shaft_solid = shaft.generate_shaft_solid()
+        blade.apply_transformations(reflect=True)
+        blade_solid = blade.generate_blade_solid(max_deg=2,
                                                  display=False,
                                                  errors=None)
         blades = []
         blades.append(blade_solid)
-        for i in range(self.number_of_blades-1):
-            rot_dir = gp_Dir(1.0,0.0,0.0)
-            origin = gp_Pnt(0.0,0.0,0.0)        
+        for i in range(n_blades-1):
+            rot_dir = gp_Dir(1.0, 0.0, 0.0)
+            origin = gp_Pnt(0.0, 0.0, 0.0)        
             rot_axis = gp_Ax1(origin, rot_dir)
             transf = gp_Trsf()
-            transf.SetRotation(rot_axis, float(i+1)*2.0*np.pi/float(self.number_of_blades));
+            transf.SetRotation(rot_axis, float(i+1)*2.0*np.pi/float(n_blades));
             transformer = BRepBuilderAPI_Transform(blade_solid, transf);
             blades.append(transformer.Shape())
         blades_combined = blades[0]
         for i in range(len(blades)-1):
-            boolean_union = BRepAlgoAPI_Fuse(blades_combined,blades[i+1])
+            boolean_union = BRepAlgoAPI_Fuse(blades_combined, blades[i+1])
             boolean_union.Build()
             if not boolean_union.IsDone():
                 raise RuntimeError('Unsuccessful assembling of blade')
             blades_combined = boolean_union.Shape()
-        boolean_union = BRepAlgoAPI_Fuse(shaft_solid, blades_combined)
+        boolean_union = BRepAlgoAPI_Fuse(self.shaft_solid, blades_combined)
         boolean_union.Build()
-        result_solid = boolean_union.Shape()
+        result_compound = boolean_union.Shape()
         section_edges = boolean_union.SectionEdges()
         sewer = BRepBuilderAPI_Sewing(1e-2)
-        sewer.Add(result_solid)
+        sewer.Add(result_compound)
         sewer.Perform()
-        sewed_full_body = sewer.SewedShape()
+        self.sewed_full_body = sewer.SewedShape()
+
+    def generate_propeller_iges(self, propeller_and_shaft, display=False):
```

Comment:
`generate_iges` for the method name (no reason to be redundant with "propeller" since the object itself is called `Propeller`)

Reply:
Right, thank you.